default.nix

```diff
@@ -358,152 +269,138 @@ let overrideCabal = pkg: f: if pkg == null then null else haskellLib.overrideCab
-      } // (if enableLibraryProfiling then {
-        mkDerivation = expr: super.mkDerivation (expr // { enableLibraryProfiling = true; });
-      } else {});
+        mkDerivation = expr: super.mkDerivation (expr // {
+          inherit enableLibraryProfiling;
+        });
+      };
     };
     haskellOverlays = import ./haskell-overlays {
       inherit
         haskellLib
         nixpkgs jdk fetchFromGitHub
-        useReflexOptimizer stage2Script;
+        useReflexOptimizer
+        hackGet;
+      inherit (nixpkgs) lib;
       androidActivity = hackGet ./android-activity;
     };
-    stage2Script = nixpkgs.runCommand "stage2.nix" {
-      GEN_STAGE2 = builtins.readFile (nixpkgs.path + "/pkgs/development/compilers/ghcjs/gen-stage2.rb");
-      buildCommand = ''
-        echo "$GEN_STAGE2" > gen-stage2.rb && chmod +x gen-stage2.rb
-        patchShebangs .
-        ./gen-stage2.rb "${sources.ghcjs-boot}" >"$out"
-      '';
-      nativeBuildInputs = with nixpkgs; [
-        ruby cabal2nix
-      ];
-    } "";
-    ghcjsCompiler = ghc.callPackage (nixpkgs.path + "/pkgs/development/compilers/ghcjs/base.nix") {
-      bootPkgs = ghc;
-      ghcjsSrc = sources.ghcjs;
-      ghcjsBootSrc = sources.ghcjs-boot;
-      shims = sources.shims;
-      stage2 = import stage2Script;
+    ghcjs8_2_2Packages = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules") {
+      ghc = ghc8_2_2.ghcjs;
+      buildHaskellPackages = ghc8_2_2.ghcjs.bootPkgs;
+      compilerConfig = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules/configuration-ghc-7.10.x.nix") { inherit haskellLib; };
+      packageSetConfig = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules/configuration-ghcjs.nix") { inherit haskellLib; };
+      inherit haskellLib;
     };
-    ghcjsPackages = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules") {
-      ghc = ghcjsCompiler;
+    ghcjs8_4_3Packages = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules") {
+      ghc = ghc8_4_3.ghcjs;
+      buildHaskellPackages = ghc8_4_3.ghcjs.bootPkgs;
       compilerConfig = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules/configuration-ghc-7.10.x.nix") { inherit haskellLib; };
       packageSetConfig = nixpkgs.callPackage (nixpkgs.path + "/pkgs/development/haskell-modules/configuration-ghcjs.nix") { inherit haskellLib; };
       inherit haskellLib;
     };
-#    TODO: Figure out why this approach doesn't work; it doesn't seem to evaluate our overridden ghc at all
-#    ghcjsPackages = nixpkgs.haskell.packages.ghcjs.override {
-#      ghc = builtins.trace "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX" ghcjsCompiler;
-#    };
-  ghcjs = (extendHaskellPackages ghcjsPackages).override {
+  ghc = ghc8_4_3;
+  ghcjs8_2_2 = (extendHaskellPackages ghcjs8_2_2Packages).override {
```

Comment:
While we're at it, it would be nice if we could switch to the nixpkgs naming scheme for all the GHC versions. I know it's technically not as precise, but the names from nixpkgs are much easier to type, and they'll be the names that new people expect when coming from nixpkgs.